app/src/main/java/org/metabrainz/mobile/activity/ArtistActivity.java

```diff
@@ -54,35 +38,56 @@
 
     private ArtistPagerAdapter pagerAdapter;
 
+    private ViewPager viewPager;
+    private TabLayout tabLayout;
+
     public void onCreate(Bundle savedInstanceState) {
-        requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
         super.onCreate(savedInstanceState);
 
-        mbid = getIntent().getStringExtra(Extra.ARTIST_MBID);
+        artistViewModel = ViewModelProviders.of(this).get(ArtistViewModel.class);
+
         setContentView(R.layout.activity_artist);
-        configurePager();
-        findViews();
-        setSupportProgressBarIndeterminateVisibility(false);
-        getSupportLoaderManager().initLoader(ARTIST_LOADER, savedInstanceState, this);
-    }
+        setSupportActionBar(findViewById(R.id.toolbar));
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-    private void configurePager() {
         pagerAdapter = new ArtistPagerAdapter(getSupportFragmentManager());
-        ViewPager pager = findViewById(R.id.pager);
-        pager.setAdapter(pagerAdapter);
-        //TabPageIndicator indicator = (TabPageIndicator) findViewById(R.id.indicator);
-        //indicator.setViewPager(pager);
-        pager.setCurrentItem(1);
-        pager.setOffscreenPageLimit(pagerAdapter.getCount() - 1);
+
+        viewPager = findViewById(R.id.pager);
+        tabLayout = findViewById(R.id.tabs);
+
+        viewPager.setVisibility(View.GONE);
+        tabLayout.setVisibility(View.GONE);
+
+        viewPager.setAdapter(pagerAdapter);
+        tabLayout.setupWithViewPager(viewPager);
     }
 
-    private void findViews() {
-        loading = findViewById(R.id.loading);
-        error = findViewById(R.id.error);
-        ratingBar = findViewById(R.id.rating);
-        tagView = findViewById(R.id.tags);
+    @Override
+    protected void onResume() {
+        super.onResume();
+        String id = getIntent().getStringExtra(Extra.ARTIST_MBID);
+        if(id != null && !id.isEmpty())
```

Comment:
```suggestion
        if(id != null && !id.isEmpty()) artistViewModel.setMBID(id);
```

`mbid` is only used here. Maybe we could get rid of it…